internal/services/iothub/iothub_device_update_account_resource.go

```diff
@@ -0,0 +1,284 @@
+				return err
+			}
+
+			resp, err := client.AccountsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			state := IotHubDeviceUpdateAccountModel{
+				Name:              id.AccountName,
+				ResourceGroupName: id.ResourceGroupName,
+				Location:          location.Normalize(model.Location),
+			}
+
+			identityValue, err := identity.FlattenLegacySystemAndUserAssignedMap(model.Identity)
+			if err != nil {
+				return fmt.Errorf("flattening `identity`: %+v", err)
+			}
+
+			if err := metadata.ResourceData.Set("identity", identityValue); err != nil {
+				return fmt.Errorf("setting `identity`: %+v", err)
+			}
+
+			if properties := model.Properties; properties != nil {
+				if properties.HostName != nil {
+					state.HostName = *properties.HostName
+				}
+
+				if properties.PublicNetworkAccess != nil {
+					state.PublicNetworkAccess = *properties.PublicNetworkAccess
+				}
+
+				if properties.Sku != nil {
+					state.Sku = *properties.Sku
+				}
+			}
+			if model.Tags != nil {
+				state.Tags = *model.Tags
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r IotHubDeviceUpdateAccountResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.IoTHub.DeviceUpdatesClient
+
+			id, err := deviceupdates.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model IotHubDeviceUpdateAccountModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.AccountsGet(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			existing := resp.Model
+			if existing == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			if existing.Properties == nil {
+				existing.Properties = &deviceupdates.AccountProperties{}
+			}
+
+			if metadata.ResourceData.HasChange("identity") {
+				identityValue, err := identity.ExpandLegacySystemAndUserAssignedMap(metadata.ResourceData.Get("identity").([]interface{}))
+				if err != nil {
+					return fmt.Errorf("expanding `identity`: %+v", err)
+				}
+				existing.Identity = identityValue
+			}
+
+			if metadata.ResourceData.HasChange("public_network_access") {
+				existing.Properties.PublicNetworkAccess = &model.PublicNetworkAccess
+			}
+
+			if metadata.ResourceData.HasChange("sku") {
+				existing.Properties.Sku = &model.Sku
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				existing.Tags = &model.Tags
+			}
+
+			existing.Properties.HostName = nil
+			existing.Properties.Locations = nil
+			existing.SystemData = nil
```

Comment:
This shouldn't need to be zeroed, ARM spec says the service is supposed to ignore R/O fields on update payloads? Needing this would indicated a bug?